tests/test_backend_gurobi.py

```diff
@@ -0,0 +1,312 @@
+        )
+        assert "foo" in simple_supply_gurobi.backend.objectives
+
+    def test_object_string_representation(self, simple_supply_gurobi):
+        assert (
+            simple_supply_gurobi.backend.variables.flow_out.sel(
+                nodes="a",
+                techs="test_supply_elec",
+                carriers="electricity",
+                timesteps="2005-01-01 00:00",
+            )
+            .item()
+            .VarName
+        ).startswith("C")
+        assert not simple_supply_gurobi.backend.variables.flow_out.coords_in_name
+
+    @pytest.mark.parametrize(
+        ("objname", "dims", "objtype", "namegetter"),
+        [
+            (
+                "flow_out",
+                {
+                    "nodes": "a",
+                    "techs": "test_supply_elec",
+                    "carriers": "electricity",
+                    "timesteps": "2005-01-01 00:00",
+                },
+                "variables",
+                "VarName",
+            ),
+            (
+                "system_balance",
+                {
+                    "nodes": "a",
+                    "carriers": "electricity",
+                    "timesteps": "2005-01-01 00:00",
+                },
+                "constraints",
+                "ConstrName",
+            ),
+        ],
+    )
+    def test_verbose_strings(
+        self, simple_supply_longnames, objname, dims, objtype, namegetter
+    ):
+        obj = simple_supply_longnames.backend._dataset[objname]
+        assert (
+            getattr(obj.sel(dims).item(), namegetter)
+            == f"{objname}[{', '.join(dims[i] for i in obj.dims)}]"
+        )
+        assert obj.attrs["coords_in_name"]
+
+    def test_verbose_strings_expression(self, simple_supply_longnames):
+        dims = {"nodes": "a", "techs": "test_supply_elec", "costs": "monetary"}
+
+        obj = simple_supply_longnames.backend.get_global_expression(
+            "cost_investment", as_backend_objs=False
+        )
+
+        assert "flow_cap[a, test_supply_elec, electricity]" in obj.sel(dims).item()
+        # parameters are not gurobi objects, so we don't get their names in our strings
+        assert "parameters[cost_interest_rate]" not in obj.sel(dims).item()
+
+        assert not obj.coords_in_name
+
+    @staticmethod
+    def _is_fixed(val):
+        return val.lb == val.ub
+
+    def test_fix_variable(self, simple_supply_gurobi):
+        simple_supply_gurobi.build(backend="gurobi", force=True)
+        simple_supply_gurobi.solve(force=True)
+        simple_supply_gurobi.backend.fix_variable("flow_cap")
+        fixed = simple_supply_gurobi.backend._apply_func(
+            self._is_fixed,
+            simple_supply_gurobi.backend.variables.flow_cap.notnull(),
+            1,
+            simple_supply_gurobi.backend.variables.flow_cap,
+        )
+        assert fixed.where(fixed.notnull()).all()
+
+        # reset
+        simple_supply_gurobi.build(backend="gurobi", force=True)
```

Comment:
Just to confirm:  is this because the fixture is shared across "class" tests?

Reply:
Also, maybe consider a function for these so that the lines are not repeated as much?